sync/blockBuffer.go

```diff
@@ -22,6 +22,16 @@ func (bb *blockBuffer) Push(b *datastore.BlockData) {
 	}
 }
 
+func (bb *blockBuffer) QuickPopFront() *datastore.BlockData {
```

Comment:
What does QuickPopFront do differently than PopFront and why was it needed?

It looks as though a different block than the front will be returned

Reply:
I ended up coming back around to this and making a quick test to confirm the two functions do return different results

```go
func TestQuick(t *testing.T) {
	bb := blockBuffer{}
	bb.Push(&datastore.BlockData{SecSinceLastBlock: 1})
	bb.Push(&datastore.BlockData{SecSinceLastBlock: 2})
	a := bb.PopFront().SecSinceLastBlock
	fmt.Println(a)

	bb = blockBuffer{}
	bb.Push(&datastore.BlockData{SecSinceLastBlock: 1})
	bb.Push(&datastore.BlockData{SecSinceLastBlock: 2})
	b := bb.QuickPopFront().SecSinceLastBlock
	fmt.Println(b)

	if a != b {
		t.Errorf("Value mismatch. %d != %d", a, b)
	}
}
```

```
=== RUN   TestQuick
2
1
--- FAIL: TestQuick (0.00s)
    blockBuffer_test.go:131: Value mismatch 2 != 1
FAIL
```

Reply:
When I was implementing the reorganization code, I was trying to find a way to speed up the return of the new head after popping a block off, which is why this code exists.

After discussing with you on RocketChat, we decided to swap to using only "PopFront" and logging the popped block instead of the new `recentBlocks` head. I will remove the now unused `QuickPopFront` function :)